data/transaction/transaction.go

```diff
@@ -102,7 +97,8 @@ func (tx *Transaction) IsInterfaceNil() bool {
 
 // GetValue returns the value of the transaction
 func (tx *Transaction) GetValue() *big.Int {
-	return tx.Value
+	val, _ := big.NewInt(0).SetString(tx.Value, 10)
+	return val
```

Comment:
this is suboptimal. Every time value is called to create a new big.int. also setString consumes processing.

Reply:
Totally agree. I'm not happy of this solution, either.

Reply:
Done, changed back so transaction.Value is *big.Int but transaction has custom methods for marshaling and unmarshaling that treats the Value field as string.